presidio-analyzer/presidio_analyzer/batch_analyzer_engine.py

```diff
@@ -127,9 +127,7 @@ def analyze_dict(
     @staticmethod
     def _validate_types(value_iterator: Iterable[Any]) -> Iterator[Any]:
         for val in value_iterator:
-            if not val:
-                continue
-            if not type(val) in (int, float, bool, str):
+            if val and not type(val) in (int, float, bool, str):
```

Comment:
there is a yield val at the bottom, are you sure you want to do it if val is empty?

Reply:
Yes, that's the issue with the previous logic. If there is no val, we still want to return a value for it. Otherwise we'll have a mismatch between input and output. For example, if we pass a list of 4 and one of the values is None, we'd like to get an output of size 4, where the output for the None is empty.

Reply:
Perhaps add this to the commit description :)

Reply:
Good idea :)

Reply:
Done.